client/state/data-layer/wpcom/sites/comments/likes/new/index.js

```diff
@@ -23,13 +25,29 @@ export const likeComment = ( { dispatch }, action ) => {
 	}, action ) );
 };
 
-export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId }, next, { like_count } ) => dispatch( local( {
-	type: COMMENTS_LIKE,
-	siteId,
-	postId,
-	commentId,
-	like_count
-} ) );
+export const updateCommentLikes = ( { dispatch }, { siteId, postId, commentId, status }, next, { like_count } ) => {
+	dispatch( local( {
+		type: COMMENTS_LIKE,
+		siteId,
+		postId,
+		commentId,
+		like_count
+	} ) );
+
+	if ( isEnabled( 'comments/undo-in-datalayer' ) && 'unapproved' === status ) {
+		const noticeId = `comment-notice-${ commentId }`;
+
+		dispatch( removeNotice( noticeId ) );
+
+		dispatch( successNotice( translate( 'Comment approved.' ), {
+			duration: 5000,
+			id: noticeId,
+			isPersistent: true,
+			button: translate( 'Undo' ),
+			onClick: () => dispatch( changeCommentStatus( siteId, postId, commentId, 'unapproved' ) )
```

Comment:
I think the approved comment disappearance is my fault actually.
To be fair though, I'd say that implementing the logic that prevents approved comments to disappear requires the actual comments data pulled from the Redux state, because it isn't easily fakeable with the CommentFaker.

Also, just to be clear: this "persistence" behaviour is only applied to "unapproved -> approved" or to any other action that moves comments between lists (e.g. "spam -> approved", etc.)?

About the whole "like+approve": since a "liked unapproved" comment doesn't really make sense, I've always worked with the impression that "undo like+approve" = "unlike+unapprove".